packages/react-relay/__tests__/ReactRelayQueryRenderer-test.js

```diff
@@ -1602,6 +1602,68 @@ describe('ReactRelayQueryRenderer', () => {
 
       expect(render.mock.calls).toHaveLength(1);
     });
+    it('check GC with single pending request', () => {
```

Comment:
what exactly are we testing here? that GC runs after the component is unmounted?

Reply:
When the queryrenderer is unmounted it performs the "dispose" of the fetcher query and then all the references are disposed. The "dispose" of the references foresee two behaviors:
* deletion of the root from the store
* schedule the GC

The GC provides for the cancellation of all entries from the record source associated with the roots no longer present.
So we're testing the actual deletion of all the roots created during the queryrenderer life cycle.